lib/appium.js

```diff
@@ -60,6 +62,18 @@ class AppiumDriver extends BaseDriver {
     let d = new InnerDriver(this.args);
     let [innerSessionId, dCaps] = await d.createSession(caps, reqCaps, curSessions);
     this.sessions[innerSessionId] = d;
+
+    // Remove the session on unexpected shutdown, so that we are in a position
+    // to open another session later on.
+    // TODO: this should be removed and replaced by a onShutdown callback.
+    d.onUnexpectedShutdown
```

Comment:
I don't want to change this for the beta, It is going to be a mess to chain `onUnexpectedShutdown` method or go into crazy class method override mode. This reveals a bigger issue, which we will need to tackle later.  https://github.com/appium/appium/issues/5537